src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -1,32 +1,39 @@
+import os
+from proto.import_all_protos import *
+from software.py_constants import MILLISECONDS_PER_SECOND
+from software.thunderscope.constants import ProtoConfigurationConstant
+import logging
 from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtCore import QTimer
 from pyqtgraph.Qt.QtWidgets import *
 from pyqtgraph import parametertree
 from google.protobuf.json_format import MessageToDict
 from thefuzz import fuzz
 from proto.import_all_protos import *
 from software.thunderscope.common import proto_parameter_tree_util
+from typing import Callable
+from PyQt6.QtWidgets import *
 
 
 class ProtoConfigurationWidget(QWidget):
-
     """Creates a searchable parameter widget that can take any protobuf,
     and convert it into a pyqtgraph ParameterTree. This will allow users
     to modify the values.
 
     """
 
+    # where we are saving the default configuration file
+
     def __init__(
-        self, proto_to_configure, on_change_callback, search_filter_threshold=60,
+        self, on_change_callback, is_yellow, search_filter_threshold=60,
```

Comment:
nit: could you add type hints? It'd be great if you could add it to the other functions missing them as well.